### ISSUE TITLE
.pdf file loses filename on download

### ISSUE THREAD

**Issue by tbsbdr** [owncloud/web#6167]
### Steps to reproduce
1.  Ensure in your browser settings, that your browser does not act as PDF-viewer
2. open demo.owncloud.com and login
3. click on a pdf-file, eg. `ownCloud Manual.pdf`
4. The download dialog shows then `7de6d7ad-60b2-4d71-81b8-273bb60d4cc8` (without extension) as the downloaded file name (current chrome or safari). On Firefox the downloaded file is just called `pdf` (without extension)

### Expected behaviour
original filename should be kept

### Actual behaviour
filename changes to a uuid `7de6d7ad-60b2-4d71-81b8-273bb60d4cc8` or `pdf` depending on the browser

![image](https://user-images.githubusercontent.com/26610733/146603541-0ce11584-0395-403e-9667-e777a552586f.png)
![image](https://user-images.githubusercontent.com/26610733/146603557-b2cb6e4f-dba8-4bc1-88df-df317c015036.png)
![image](https://user-images.githubusercontent.com/26610733/146603575-adee8739-703f-4c4f-9fcc-a26478109321.png)

**Comment by fschade** [comment]
@tbsbdr there is only one way to workaround the described issue.
If we decide to always download a pdf (instead of opening it in a new tab and let the browser handle the pdf rendering) we can use a workaround and overwrite the download name.

the uid you see here is how the browser internally stores the blob and references it. We can, as described above set the download name (https://www.w3schools.com/tags/att_a_download.asp) manual but the side-effect is that the browser is always downloading the file.

another solution could be to write a pdf viewer app (https://github.com/mozilla/pdf.js) and then let the user decide what should happen. Please let me know how we proceed and what you prefer.

**Comment by tbsbdr** [comment]
yep, the browser behaviour (=user setting) should be respected - i.e. no "download" attribute in the link.
"As a user who left-clicks on [myDocument.pdf ](https://www.rundfunkbeitrag.de/e175/e5237/Buergerinnen_und_Buerger_Kontoaenderung_0104.pdf) in the fileslist, I expect that the same thing happens like on any other website when I click on a pdf."

If the user chooses righclick -> download the a download should start.

**Comment by fschade** [comment]
@tbsbdr as discussed i tried the preSigned url which worked expect that the backend uses "Content-Disposition: attachment".

This force downloads the requested file.

The only possible solution i can think of is to change the webdav behavior in oc10 and ocis to support switching "Content-Disposition: inline"

**Comment by tbsbdr** [comment]
`inline` would also be "wrong" if this means, that the backend and not the browser defines the behaviour in this usecase. would it be possible to just don't send the content disposition header so that the decision is up to the browser?

1. minimum goal is to not loose the filename
2. cherry on top: let the browser decide about download or view

**Comment by kulmann** [comment]
I'd like to follow up here and bring @C0rby to the game.

Is it a good idea to provide a `downloadUrl` and an `inlineUrl`, both signed, backend side? With `Content-Disposition: attachment`  and `Content-Disposition:inline` respectively. And of course for both authenticated contexts and public links. Then the implementation in web would be super slim.

**Comment by C0rby** [comment]
I think we need to add this to the next sprint. This needs some thinking through first. I would like to have the signed URL only on-demand because the signing is actually a bit of a computational heavy operation and doing it on every propfind isn't very efficient because you don't need the signed URLs every time.

**Comment by diocas** [comment]
This issue seems a bit more generic than what I understood from our discussion this. But let me add our 2 cents...

From our point of view, the current preview implementation also has issues because we cannot bookmark/reuse the url (and is non understandable by users as well). At the same time, we lose the branding which might be inconsistent with other extensions.
But the biggest issue, and one that we received many complains from users, is that there is no feedback. Meaning that if we try to open a large PDF, nothing happens because the browser needs to download it first, and only then opens the new tab with the preview.

I'm also not a big fan of having signed urls, that become invalid after expiration. So the idea of putting it inside an iframe, where we show a pretty normal app url, with this logic hidden is better.

We've tried to do the iframe method, and still use the browser native renderer. This solved all the issues I mentioned but it created a major one that I haven't been able to fix: the presentation mode no longer works! (even with `allowfullscreen`).
Maybe the usage of a library could be reconsidered as well?

**Comment by kulmann** [comment]
@diocas I like your solution a lot. Could you provide a PR so that we can check as well if we manage to get the fullscreen support running? In any case, using an iframe to bypass the issues you mentioned and still use the browsers native pdf support is just awesome.

**Comment by tbsbdr** [comment]
> But the biggest issue, and the one that caused many complains from users, is that there is no (loading) feedback. Meaning that if we try to open a large PDF, nothing happens because the browser needs to download it first, and only then opens the new tab with the preview. And users kept clicking, which then would do another download and eventually replace the current open tab.

@diocas I made a "Add progress indicator for ui-interactions" issue here https://github.com/owncloud/web/issues/6183 ticket is currently not prioritised but I'll discuss with benedikt if we give it a higher prio.

**Comment by lookacat** [comment]
@tbsbdr Seems to be working now..
